test/tests/input_fail.py

```diff
@@ -0,0 +1,11 @@
+# should_error
+
+# test failure case to make sure the exception types match
+import StringIO
+import sys
+
+orig_stdin = sys.stdin
+sio = StringIO.StringIO("abcd")
+sys.stdin = sio
+print repr(input())
+sys.stdin = orig_stdin
```

Comment:
Minor stylistic note, but we should try to avoid `# should_error` tests if we can (and usually we can now).  those have bitten us in the past since if we add a typo to the test, all of a sudden we're only testing whether we detect the typo the same way that CPython does.

to verify if an exception is thrown usually we can do something like
```python
try:
    print repr(input())
except SyntaxError:
    print "caught expected syntax error"
```